0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -56,3 +52,115 @@ All ILP Packets MUST be returned with the HTTP status code `200: OK`.
+
+Because tokens in this profile do not inherently contain information about the identity of the caller, requests MUST contain an additional HTTP request-header named `Auth-Principal`.
+ 
+This extra header allows for the identity of the authentication request to be separated from authentication token itself, which reduces computational overhead as well as data-management complexity (e.g., implementations do not need to create data-store indexes using derivations of tokens for lookup purposes).
+
+Depending on the implementation, the value of this header SHOULD represent an `ILP Account Id`, but future implementations MAY make this represent an `ILP Peer Id` or some other value.
+
+##### Example Usage
+An example shared-secret in this profiles is `HEiMCp0FoAC903QHueY89gAWJHo/izaBnJU8/58rlSI=`. This shared secret is passed as an `Authorization` header in each HTTP request, using the Bearer token scheme, along with an `Auth-Principal` header like this: 
+
+```
+Auth-Principal: alice-usd-123
+Authorization: Bearer HEiMCp0FoAC903QHueY89gAWJHo/izaBnJU8/58rlSI=
+``` 
+
+Implementations MAY support this profile, but SHOULD consider it for development purposes only.
+ 
+##### Trade-off Summary
+* **Pros**
+  * The simplest, most usable Authentication profile -- just a shared-secret with _at least_ 32 bytes and an identity header.
+  * Very little processing time to verify a token (just a simple value comparison or, for more security, an HMAC-SHA256 computation).
+
+* **Cons**
+  * The shared-secret is transmitted "on the wire" for every request, increasing the chances that it might be intercepted by a compromised TLS session (e.g., a [MITM attack](https://en.wikipedia.org/wiki/Man-in-the-middle_attack)); a TLS termination endpoint (e.g., a Load Balancer); or logged by an internal system during transit.
+  * The shared-secret itself never expires, so if an implementation neglects to rotate the secret with its peer, this token will likely be very long-lived. This increases the chance of compromise by an attacker, and means compromised usage of this type of token could go undetected for very longer periods of time.
+  * Requires out-of-band communication for both peers to agree upon the shared secret.
+
+#### `JWT_HS_256` Authentication Profile
+This profile allows two ILP nodes to utilize a previously agreed-upon shared-secret, but then derive an [RFC-7519](https://tools.ietf.org/html/rfc7519) compliant JWT token in order to perform actual authentication.
+
+##### JWT Claims
+In order to be considered a valid JWT for this profile, the signed JWT MUST contain a `sub` (subject) claim containing the identifier of the "principal" that the token authenticates. Depending on the implementation, the value of this claim SHOULD represent an `ILP Account Id`, but future implementations MAY make this represent an `ILP Peer Id` or some other value.
```

Comment:
As above I'd drop this because there is a lot of implicit meaning in those names which we don't (need to) define:

> Depending on the implementation, the value of this claim SHOULD represent an `ILP Account Id`, but future implementations MAY make this represent an `ILP Peer Id` or some other value.

IMO what you have is enough, its standard authentication/identity terminology to simply say the "principal".

Reply:
This is now [removed](https://github.com/interledger/rfcs/pull/531/commits/24bca61866b9e7f4004fdd546cfc57f7515a5f7e).